src/core/public/chrome/ui/header/recent_items.tsx

```diff
@@ -94,46 +147,132 @@ export const RecentItems = ({
+            ...recentAccessItem.meta,
+            updatedAt: moment(obj?.updated_at).valueOf(),
+            workspaceName: findWorkspace?.name,
+          };
+        });
+        setDetailedSavedObjects(formatDetailedSavedObjects);
+      });
+    }
+  }, [recentlyAccessedItems, http, workspaceList]);
+
+  const selectedRecentsItems = useMemo(() => {
+    return detailedSavedObjects.slice(0, Number(recentsRadioIdSelected.split('recent_')[1]));
+  }, [detailedSavedObjects, recentsRadioIdSelected]);
+
   return (
     <EuiPopover
-      button={button}
+      button={recentButton}
       isOpen={isPopoverOpen}
       closePopover={() => {
         setIsPopoverOpen(false);
       }}
       anchorPosition="downCenter"
       repositionOnScroll
       initialFocus={false}
-      panelPaddingSize="s"
+      panelPaddingSize="m"
     >
       {renderBreadcrumbs}
       <EuiSpacer size="s" />
+      <EuiPanel
+        hasShadow={false}
+        hasBorder={false}
+        paddingSize="none"
+        style={{ maxHeight: '35vh', overflow: 'auto' }}
+      >
+        <EuiTitle size="xxs">
+          <h4>Recent</h4>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+        {selectedRecentsItems.length > 0 ? (
+          <EuiListGroup flush={true} gutterSize="s">
+            {selectedRecentsItems.map((item) => (
+              <EuiListGroupItem
+                onClick={() => handleItemClick(item.link)}
+                key={item.link}
+                style={{ padding: '1px' }}
+                label={
+                  <>
+                    <EuiIcon
+                      style={{ marginRight: widthForRightMargin }}
+                      type={item.meta.icon || 'apps'}
+                    />
+                    {item.label}
+                    {item.workspaceName ? (
+                      <EuiTextColor color="subdued">({item.workspaceName})</EuiTextColor>
+                    ) : null}
+                  </>
+                }
+                color="text"
+                size="s"
+              />
+            ))}
+          </EuiListGroup>
+        ) : (
+          <EuiText color="subdued">No recently viewed items</EuiText>
+        )}
+        <EuiSpacer size="s" />
 
-      <EuiTitle size="xxs">
-        <h4>Recents</h4>
-      </EuiTitle>
-      {items.length > 0 ? (
-        <EuiListGroup>
-          {items.map((item) => (
-            <EuiListGroupItem
-              onClick={() => handleItemClick(item.link)}
-              key={item.link}
-              label={
-                <>
-                  {item.label}
-                  {item.workspaceName ? (
-                    <EuiTextColor color="subdued">({item.workspaceName})</EuiTextColor>
-                  ) : null}
-                </>
-              }
-              color="text"
-              size="s"
-            />
-          ))}
-        </EuiListGroup>
-      ) : (
-        <EuiText color="subdued">No recently viewed items</EuiText>
-      )}
+        <EuiPopover
```

Comment:
Nit: we could extract preference popover into a new component, to make two different components isolate.

Reply:
Thank you, updated.